packages/platforms/delta/armel/arm-delta-ag6248c/src/arm_delta_ag6248c/module/src/sysi.c

```diff
@@ -0,0 +1,290 @@
+		return "unknow";
+}
+
+int
+onlp_sysi_platform_set(const char* platform)
+{ 
+    if(strstr(platform,"arm-delta-ag6248c-r0")) {
+        platform_id = PLATFORM_ID_POWERPC_DELTA_AG6248C_R0;
+        return ONLP_STATUS_OK;
+    }
+    if(strstr(platform,"arm-delta-ag6248c-poe-r0")) {
+        platform_id = PLATFORM_ID_POWERPC_DELTA_AG6248C_POE_R0;
+        return ONLP_STATUS_OK;
+    }
+	AIM_LOG_ERROR("No support for platform '%s'", platform);
+    return ONLP_STATUS_E_UNSUPPORTED;
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{ 
+    int   v;
+    
+	v = i2c_devname_read_byte("CPLD", 0X07);
+	
+    pi->cpld_versions = aim_fstrdup("%d", v);
+
+    return 0;
+}
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{ 	
+    uint8_t* rdata = aim_zmalloc(256);
+	int fd,rc_size;
+	char  fullpath[20] = {0};
+	
+	sprintf(fullpath, "/dev/mtd7");
+	
+	fd=open(fullpath,O_RDWR);
+	
+	if(fd<0){
+		aim_free(rdata);
+		return ONLP_STATUS_E_INTERNAL ;
+	}
+		
+	rc_size=read(fd,rdata,256);
+	
+	if(rc_size<0||rc_size!=256){
+		aim_free(rdata);
+		return ONLP_STATUS_E_INTERNAL ;
+	}
+			
+    *data = rdata;
+
+    return ONLP_STATUS_OK;
+
+	
+}
+
+void
+onlp_sysi_onie_data_free(uint8_t* data)
+{ 
+    aim_free(data);
+}
+
+
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{ 
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 1 Thermal sensors on the chassis */
+    for (i = 1; i <= CHASSIS_THERMAL_COUNT; i++) {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* LEDs on the chassis */
+    for (i = 1; i <= chassis_led_count(); i++) {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 1 Fans on the chassis */
+    for (i = 1; i <= chassis_fan_count(); i++) {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    /* 2 PSUs on the chassis */
+    for (i = 1; i <= CHASSIS_PSU_COUNT; i++) {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_platform_manage_fans(void)
+{ 
+
+    int rc;
+    onlp_thermal_info_t ti1;
+    onlp_thermal_info_t ti2;
+    int mtemp=0;
+    int new_rpm=0;
+
+    if (chassis_fan_count() == 0) {
+        return ONLP_STATUS_E_UNSUPPORTED;
+    }
+  
+    /* Get temperature */
+    rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(1), &ti1);
+	
+    if (rc != ONLP_STATUS_OK) {
+        return rc;
+    }
+        
+    rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(2), &ti2);
+	
+    if (rc != ONLP_STATUS_OK) {
+        return rc;
+    }
+             
+    mtemp=(ti1.mcelsius+ti2.mcelsius)/2;
+    /* Bring fan speed according the temp
+     */
+    if(mtemp<50000)
+        new_rpm=FAN_IDLE_RPM;
+    else if((mtemp>=55000)&&(mtemp<60000))
+        new_rpm=FAN_LEVEL1_RPM;
+    else if((mtemp>=65000)&&(mtemp<70000))
+        new_rpm=FAN_LEVEL2_RPM;
+    else if(mtemp>=75000)
+        new_rpm=FAN_LEVEL3_RPM;
+    else{
+        return ONLP_STATUS_OK;
+   }
+       
+    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(1),new_rpm);
```

Comment:
onlp_fan_rpm_set, not onlp_fani_rpm_set.

Reply:
the function "onlp_fani_rpm_set" is our implement, not the "onlp_fan_rpm_set" , when we want get the fan, the the function "onlp_thermali_info_get"  maybe been good , not the function "onlp_thermal_info_get" . the function "onlp_thermali_info_get" is the underlying  interface, what's more, all the other is realized as the same.

Reply:
as the similar structrue between ag6248c and ag6248c_poe, so the ONLPSIM of them is the same.
so the platform ag6248c and ag6248c_poe using the common one , the common name is the arm_delta_ag6248c. therefore, the ONLPSIM files can not be converted.

Reply:
See my previous comments.